python/src/spark_rapids_ml/regression.py

```diff
@@ -524,25 +608,30 @@ def _create_pyspark_model(self, result: Row) -> "LinearRegressionModel":
     def _enable_fit_multiple_in_single_pass(self) -> bool:
         return True
 
+    def _supportsTransformEvaluate(self, evaluator: Evaluator) -> bool:
+        return True if isinstance(evaluator, RegressionEvaluator) else False
+
 
 class LinearRegressionModel(
     LinearRegressionClass,
     _CumlModelWithPredictionCol,
     _LinearRegressionCumlParams,
+    _RegressionModelEvaluationMixIn,
 ):
     """Model fitted by :class:`LinearRegression`."""
 
     def __init__(
         self,
-        coef_: List[float],
-        intercept_: float,
+        coef_: Union[List[float], List[List[float]]],
+        intercept_: Union[float, List[float]],
```

Comment:
Wondering if just assuming lists here would simplify things elsewhere, where the single model is just represented as a list of one item.

Reply:
Wow, Good suggestion. I will file a followup PR to handle it not only for LinearRegression but also for RandomForest.